airflow/providers/google/cloud/hooks/vertex_ai/batch_prediction_job.py

```diff
@@ -241,6 +244,159 @@ def create_batch_prediction_job(
+        :param labels: Optional. The labels with user-defined metadata to organize your BatchPredictionJobs.
+            Label keys and values can be no longer than 64 characters (Unicode codepoints), can only contain
+            lowercase letters, numeric characters, underscores and dashes. International characters are
+            allowed. See https://goo.gl/xmQnxf for more information and examples of labels.
+        :param encryption_spec_key_name: Optional. The Cloud KMS resource identifier of the customer managed
+            encryption key used to protect the job. Has the form:
+            ``projects/my-project/locations/my-region/keyRings/my-kr/cryptoKeys/my-key``. The key needs to be
+            in the same region as where the compute resource is created.
+            If this is set, then all resources created by the BatchPredictionJob will be encrypted with the
+            provided encryption key.
+            Overrides encryption_spec_key_name set in aiplatform.init.
+        :param create_request_timeout: Optional. The timeout for the create request in seconds.
+        :param batch_size: Optional. The number of the records (e.g. instances)
+            of the operation given in each batch
+            to a machine replica. Machine type, and size of a single record should be considered
+            when setting this parameter, higher value speeds up the batch operation's execution,
+            but too high value will result in a whole batch not fitting in a machine's memory,
+            and the whole operation will fail.
+            The default value is same as in the aiplatform's BatchPredictionJob.
+        """
+        self._batch_prediction_job = BatchPredictionJob.submit(
+            job_display_name=job_display_name,
+            model_name=model_name,
+            instances_format=instances_format,
+            predictions_format=predictions_format,
+            gcs_source=gcs_source,
+            bigquery_source=bigquery_source,
+            gcs_destination_prefix=gcs_destination_prefix,
+            bigquery_destination_prefix=bigquery_destination_prefix,
+            model_parameters=model_parameters,
+            machine_type=machine_type,
+            accelerator_type=accelerator_type,
+            accelerator_count=accelerator_count,
+            starting_replica_count=starting_replica_count,
+            max_replica_count=max_replica_count,
+            generate_explanation=generate_explanation,
+            explanation_metadata=explanation_metadata,
+            explanation_parameters=explanation_parameters,
+            labels=labels,
+            project=project_id,
+            location=region,
+            credentials=self.get_credentials(),
+            encryption_spec_key_name=encryption_spec_key_name,
+            create_request_timeout=create_request_timeout,
+            batch_size=batch_size,
+        )
+        return self._batch_prediction_job
```

Comment:
Yes and no. Despite saving the job within the hook, we should return the submitted job in the `submit_batch_prediction_job` method, because it is useful for users if they need a job instance for further usage. In this case they don't have to invoke any another method.

Reply:
Then maybe we don't need `self` at all here? 

Except for `def cancel_batch_prediction_job()` I don't see this attribute being re-used in the other hook methods. The same result can be achieved with the [API method](https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform_v1beta1.services.job_service.JobServiceClient#google_cloud_aiplatform_v1beta1_services_job_service_JobServiceClient_cancel_batch_prediction_job).

Reply:
I see your point, I agree.